com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -105,11 +106,25 @@ internal struct AgentAction
         "docs/Learning-Environment-Design-Agents.md")]
     [Serializable]
     [RequireComponent(typeof(BehaviorParameters))]
-    public abstract class Agent : MonoBehaviour
+    public abstract class Agent : MonoBehaviour, ISerializationCallbackReceiver
     {
         IPolicy m_Brain;
         BehaviorParameters m_PolicyFactory;
 
+        /// This code is here to make the upgrade path for users using maxStep
```

Comment:
Can you maybe wrap this code into a 
```csharp
#region 

#endregion 
```

Reply:
Yamato is still acting up, maybe after this gets into the release I can add the regions.

Reply:
I don't have strong feelings on it, but the C# style guide says `Use of #region is _always_ disallowed.`